mod/state-transition/pkg/core/state_processor_genesis.go

```diff
@@ -102,7 +97,18 @@ func (sp *StateProcessor[
 	}
 
 	for _, deposit := range deposits {
-		if err := sp.processDeposit(st, deposit); err != nil {
+		prevDepositIndex, err := st.GetEth1DepositIndex()
```

Comment:
the reason deposit checks have been moved here is that for non-genesis blocks we already validate deposits by comparing them with deposits in store.
So we check genesis deposits are in the right sequence here (no other checks are needed, other than signature, which is done later on) and we simplify the `processDeposit` function